src/dotnet/ToolPackage/LockFileMatcher.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+
+namespace Microsoft.DotNet.ToolPackage
+{
+    internal class LockFileMatcher
+    {
+        private static string[] _pathInLockFilePathInArray;
+        private static string[] _entryPointPathInArray;
+
+        internal static bool MatchesFile(string pathInLockFile, string entryPoint)
```

Comment:
From just reading the code without knowing more context, this class seems confusing.  `MatchesFile` is a pretty generic name and doesn't help me understand the code.

I would recommend adding comments describing what the MatchesFile method does, and the context in which it does it.